fast64_internal/f3d/f3d_material.py

```diff
@@ -1271,467 +1334,380 @@ def update_node_values_of_material(material, context):
+    textureSettings: bpy.types.ShaderNodeGroup = nodes["TextureSettings"]
 
-    shiftNode.outputs[0].default_value = shift
-    scaleNode.outputs[0].default_value = scale * uvBasisScale
+    desired_group = bpy.data.node_groups["TextureSettings_Lite"]
+    if (material.f3d_mat.tex0.tex and not material.f3d_mat.tex0.autoprop) or (
+        material.f3d_mat.tex1.tex and not material.f3d_mat.tex1.autoprop
+    ):
+        desired_group = bpy.data.node_groups["TextureSettings_Advanced"]
+    if textureSettings.node_tree is not desired_group:
+        textureSettings.node_tree = desired_group
 
 
 def setAutoProp(fieldProperty, pixelLength):
-    fieldProperty.low = 0
-    fieldProperty.high = pixelLength - 1
     fieldProperty.mask = math.ceil(math.log(pixelLength, 2) - 0.001)
-    # fieldProperty.mask = 0
     fieldProperty.shift = 0
+    fieldProperty.low = 0
+    fieldProperty.high = pixelLength
+    if fieldProperty.clamp and fieldProperty.mirror:
+        fieldProperty.high *= 2
+    fieldProperty.high -= 1
 
 
-def update_tex_values_field_v2(self, fieldProperty, pixelLength, uvBasisScale, scale, autoprop, texIndex, field):
-
-    fieldIndex = 0 if field == "S" else 1
-    pixelLengthAxis = pixelLength[fieldIndex]
-
+def set_texture_size(self, tex_size, tex_index):
     nodes = self.node_tree.nodes
-    clampNode = nodes["Tex " + str(texIndex) + " Clamp"]
-    mirrorNode = nodes["Tex " + str(texIndex) + " Mirror"]
-    normHalfPixelNode = nodes["Tex " + str(texIndex) + " Normalized Half Pixel"]
-    normLNode = nodes["Tex " + str(texIndex) + " Normalized L"]
-    normHNode = nodes["Tex " + str(texIndex) + " Normalized H"]
-    normMaskNode = nodes["Tex " + str(texIndex) + " Normalized Mask"]
-    shiftNode = nodes["Tex " + str(texIndex) + " Shift"]
-    scaleNode = nodes["Tex " + str(texIndex) + " Scale"]
+    uv_basis: bpy.types.ShaderNodeGroup = nodes["UV Basis"]
+    inputs = uv_basis.inputs
 
-    clampNode.inputs[fieldIndex].default_value = 1 if fieldProperty.clamp else 0
-    mirrorNode.inputs[fieldIndex].default_value = 1 if fieldProperty.mirror else 0
-    normHalfPixelNode.inputs[fieldIndex].default_value = 1 / (2 * pixelLengthAxis)
+    inputs[f"{tex_index} S TexSize"].default_value = tex_size[0]
+    inputs[f"{tex_index} T TexSize"].default_value = tex_size[1]
 
-    if autoprop:
-        setAutoProp(fieldProperty, pixelLengthAxis)
 
-    normLNode.inputs[fieldIndex].default_value = fieldProperty.low / pixelLengthAxis * (-1 if field == "T" else 1)
-    normHNode.inputs[fieldIndex].default_value = (fieldProperty.high + 1) / pixelLengthAxis
-    normMaskNode.inputs[fieldIndex].default_value = (
-        (2**fieldProperty.mask) / pixelLengthAxis if fieldProperty.mask > 0 else 0
-    )
-    shiftNode.inputs[fieldIndex].default_value = fieldProperty.shift
-    scaleNode.inputs[fieldIndex].default_value = scale[fieldIndex] * uvBasisScale[fieldIndex]
+def round_10_2(val: float):
+    return float(int(val * 4)) / 4.0
```

Comment:
that's not rounding, it's truncating, so why "round"?
also
```suggestion
    return int(val * 4) / 4
```
dividing integers in python produces floats when needed (integer division is `//`)

Reply:
> that's not rounding, it's truncating, so why "round"?

i can rename it to `trunc_10_2`

Reply:
renamed it, if you like the new name please resolve!